pkg/drift/terraform/terraform.go

```diff
@@ -0,0 +1,260 @@
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMBindingForOrg(instances []ResourceInstance) []*iam.AssetIAM {
+	var iams []*iam.AssetIAM
+	for _, i := range instances {
+		for _, m := range i.Attributes.Members {
+			iams = append(iams, &iam.AssetIAM{
+				Member:       m,
+				Role:         i.Attributes.Role,
+				ResourceID:   p.organizationID,
+				ResourceType: assets.Organization,
+			})
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMBindingForFolder(instances []ResourceInstance) []*iam.AssetIAM {
+	var iams []*iam.AssetIAM
+	for _, i := range instances {
+		for _, m := range i.Attributes.Members {
+			parentID, parentType := p.maybeFindGCPAssetIDAndType(i.Attributes.Folder)
+			iams = append(iams, &iam.AssetIAM{
+				Member:       m,
+				Role:         i.Attributes.Role,
+				ResourceID:   parentID,
+				ResourceType: parentType,
+			})
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMBindingForProject(instances []ResourceInstance) []*iam.AssetIAM {
+	var iams []*iam.AssetIAM
+	for _, i := range instances {
+		for _, m := range i.Attributes.Members {
+			parentID, parentType := p.maybeFindGCPAssetIDAndType(i.Attributes.Project)
+			iams = append(iams, &iam.AssetIAM{
+				Member:       m,
+				Role:         i.Attributes.Role,
+				ResourceID:   parentID,
+				ResourceType: parentType,
+			})
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMMemberForOrg(instances []ResourceInstance) []*iam.AssetIAM {
+	iams := make([]*iam.AssetIAM, len(instances))
+	for x, i := range instances {
+		iams[x] = &iam.AssetIAM{
+			Member:       i.Attributes.Member,
+			Role:         i.Attributes.Role,
+			ResourceID:   p.organizationID,
+			ResourceType: assets.Organization,
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMMemberForFolder(instances []ResourceInstance) []*iam.AssetIAM {
+	iams := make([]*iam.AssetIAM, len(instances))
+	for x, i := range instances {
+		parentID, parentType := p.maybeFindGCPAssetIDAndType(i.Attributes.Folder)
+		iams[x] = &iam.AssetIAM{
+			Member:       i.Attributes.Member,
+			Role:         i.Attributes.Role,
+			ResourceID:   parentID,
+			ResourceType: parentType,
+		}
+	}
+	return iams
+}
+
+func (p *Parser) parseIAMMemberForProject(instances []ResourceInstance) []*iam.AssetIAM {
+	iams := make([]*iam.AssetIAM, len(instances))
+	for x, i := range instances {
+		parentID, parentType := p.maybeFindGCPAssetIDAndType(i.Attributes.Project)
+		iams[x] = &iam.AssetIAM{
+			Member:       i.Attributes.Member,
+			Role:         i.Attributes.Role,
+			ResourceID:   parentID,
+			ResourceType: parentType,
+		}
+	}
+	return iams
+}
+
+func (p *Parser) maybeFindGCPAssetIDAndType(ID string) (string, string) {
+	var assetID string
+	var assetType string
+	asset := p.findGCPAsset(ID)
+	if asset == nil {
+		assetID = UnknownParentID
+		assetType = UnknownParentType
+	} else {
+		assetID = asset.ID
+		assetType = asset.NodeType
+	}
+	return assetID, assetType
```

Comment:
This seems perhaps unnecessarily complex. It could be 

```
    asset := p.findGCPAsset(ID)
    if asset == nil {
        return UnknownParentID, UnknownParentType
    }
    return asset.ID, asset.NodeType
```